### PR TITLE
Work from Imran to verify Pawn moves

### PR DESCRIPTION
**Pull request by goofballLogic**
Hi @YorgunKelebek, this work was completed by @aoshi321 but wanted to run it past you to verify we have implemented correctly (from a Chess perspective I mean...)

### CODE REVIEW

File: src/game/piece_move.rs [hunk 1]
```diff
@@ -0,0 +1,138 @@
+use super::piece_type::*;
+
+fn validate_moves(moves: Vec<(u8, u8)>) -> Vec<(u8, u8)> {
+    let mut valid_moves: Vec<(u8, u8)> = [].to_vec();
+    for m in moves.iter() {
+        let (m1, m2) = *m;
+        if (m1 > 1 && m1 < 8) && (m2 < 8) {
```
Comment (YorgunKelebek) [inline src/game/piece_move.rs]: Please review this logic to ensure moves for all piece types are validated correctly.

Comment (goofballLogic) [reply src/game/piece_move.rs]: It's just the pawn so far but now that I'm looking at it I'm wondering about `m1>1`. Will ponder this algorithm a bit more

Comment (goofballLogic) [reply src/game/piece_move.rs]: yeah i found a bug in this implementation, so I've added some more tests (one of which now breaks). I'll send this back to @aoshi321 . Thanks for catching!

Comment (goofballLogic) [reply src/game/piece_move.rs]: this is now resolved so I'm merging

**Pull request merged**